core/src/main/java/com/hevelian/identity/core/TenantService.java

```diff
@@ -80,6 +93,16 @@ public Tenant getTenant(String tenantDomain) throws TenantNotFoundByDomainExcept
     return tenant;
   }
 
+  public BufferedImage getTenantLogo(String tenantDomain) throws TenantNotFoundByDomainException, IOException {
```

Comment:
There is a super small chance that IO exception will be thrown. There are no any reasons for it in your code. This checked IO exception should not be a part of this public service method. Try to convert it to runtime or read it from bytes without streams.